packages/api/cms-api/src/user-permissions/entities/user-content-scopes.entity.ts

```diff
@@ -0,0 +1,27 @@
+import { ArrayType, Embeddable, Embedded, Entity, PrimaryKey, Property } from "@mikro-orm/core";
+import { Field, ObjectType } from "@nestjs/graphql";
+
+@ObjectType()
+@Entity()
+export class UserContentScopes {
+    @Field()
+    @PrimaryKey()
+    @Property()
+    userId: string;
+
+    @Embedded(() => UserContentScope, { array: true })
+    @Field(() => [UserContentScope])
+    scopes: UserContentScope[] = [];
```

Comment:
How would this scope schema work for intersections? 

For instance, a user may be allowed for domain Main in English and domain Secondary in German. The scopes here would be `{ scope: "domain", values: ["Main", "Secondary"] }, { scope: "language", values: ["EN", "DE"] }` if I understand it correctly? How could we ensure that the user cannot access domain Main in German?

Reply:
Solved with https://github.com/vivid-planet/comet/pull/1223/commits/10c0abf40e7966f5af44f6083aed6acb24aac0f2